src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2221,8 +2221,9 @@ public IAsyncResult BeginSend(byte[] buffer, int offset, int size, SocketFlags s
         {
             ThrowIfDisposed();
             ValidateBufferArguments(buffer, offset, size);
-
-            return TaskToApm.Begin(SendAsync(new ReadOnlyMemory<byte>(buffer, offset, size), socketFlags, default).AsTask(), callback, state);
+            var task = SendAsync(new ReadOnlyMemory<byte>(buffer, offset, size), socketFlags, default).AsTask();
+            task.ThrowIfFailedSynchronously();
```

Comment:
This only addresses the concern for SendAsync because it's a single operation.  It doesn't address cases where we're now using an async method to compose over multiple operations, like BeginConnect(IPAddress[], ...).  It used to be that this would call BeginXx on the first address in the array and propagate that exception synchronously if one occurred.  But we've cleaned up all of that code, and now all of the connection attempts to each IPAddress in the array are part of a loop in an async method.  This proposed workaround doesn't help with such cases.